src/state.py

```diff
@@ -270,6 +285,9 @@ def from_charm(cls, charm: ops.CharmBase) -> "State":
             deprecated_agent_meta_map = _get_agent_meta_map_from_relation(
                 charm.model.relations[DEPRECATED_AGENT_RELATION], charm.app.name
             )
+            is_auth_proxy_integrated = _is_auth_proxy_integrated(
+                charm.model.get_relation(AUTH_PROXY_RELATION)
+            )
```

Comment:
There may be a problem here because the relational data is set during `_on_auth_proxy_relation_joined`. However, the function `from_charm` is called in `JenkinsK8sOperatorCharm.__init__`, which occurs before `_on_auth_proxy_relation_joined`.